### ISSUE TITLE
OperationalError: no such column

### ISSUE THREAD

**Issue by IperGiove** [piccolo-orm/piccolo#762]
Hi, 
I am trying to update a row of a table Band, doing a where query over a foreign key. 

`await Band.update({Band.name: "test"}).where(Band.manager.id==1)`
 
I got the error:
 OperationalError: no such column: band$manager.id
 
```
lib/python3.11/site-packages/aiosqlite/core.py:102, in Connection.run(self)
    100 try:
    101     LOG.debug("executing %s", function)
--> 102     result = function()
    103     LOG.debug("operation %s completed", function)
    105     def set_result(fut, result):

OperationalError: no such column: band$manager.id


```

**Comment by dantownsend** [comment]
That should work OK. Have you double checked the tables exist in the database using something like pgAdmin?

**Comment by IperGiove** [comment]
Yes the tables exist, I used the command `piccolo playground run`, indeed when i do like this everything works fine:

```
In [1]: manager = await Manager.objects().where(Manager.id==1).first()
   ...: await Band.update({Band.name: "test"}).where(Band.manager==manager)
Out[1]: []

In [3]: await Band.select()
Out[3]: 
[{'id': 1, 'name': 'test', 'manager': 1, 'popularity': 1000},
 {'id': 2, 'name': 'Rustaceans', 'manager': 2, 'popularity': 500},
 {'id': 3, 'name': 'C-Sharps', 'manager': 3, 'popularity': 700}]
```

**Comment by dantownsend** [comment]
Ah, yes - I remember now. We don't currently allow joins in the where clauses when using update.

The solution for now is to do it in two steps:

```
band_ids = await Band.select(Band.id).where(Band.manager.id == 1).output(as_list=True)
await Band.update({Band.name: "test"}).where(Band.id.is_in(band_ids))
```

Sorry about that.

**Comment by IperGiove** [comment]
Ok, thank you. However, it would make sense to use the join in the where even when there is an update, do you plan to enable it in the future?

**Comment by dantownsend** [comment]
Yes, it's definitely on the roadmap. The syntax for joins is different for update queries, which is why it doesn't currently work.

**Comment by dantownsend** [comment]
Fixed in `0.106.0`.
